frontend/src/DeviceDetailsPage/__tests__/DeleteDeviceButton.test.jsx

```diff
@@ -0,0 +1,100 @@
+import React from "react";
+import {
+  Button,
+  Dialog,
+  DialogTitle,
+  DialogContentText,
+  DialogActions
+} from "@material-ui/core";
+
+import Enzyme from "enzyme";
+import Adapter from "enzyme-adapter-react-16";
+import { afterEach, describe, expect, jest, it } from "@jest/globals";
+import axios from "axios";
+
+import DeleteDeviceButton from "../DeleteDeviceButton";
+
+Enzyme.configure({ adapter: new Adapter() });
+jest.mock("axios");
+
+const mockHistoryPush = jest.fn();
+jest.mock("react-router-dom", () => ({
+  useHistory: () => ({
+    push: mockHistoryPush
+  })
+}));
+
+describe("DeleteButton", () => {
+  let wrapper;
+
+  const setOpen = jest.fn();
+  const handleClick = jest.spyOn(React, "useState");
+
+  beforeEach(() => {
+    handleClick.mockImplementation((open) => [open, setOpen]);
+    wrapper = Enzyme.shallow(
+      <DeleteDeviceButton deviceType="encoder" deleteId="1:10:111:999" />
+    );
+  });
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("Should have 3 total buttons and one dialog", () => {
+    expect(wrapper.find(Button)).toHaveLength(3);
+    expect(wrapper.find(Dialog)).toHaveLength(1);
+  });
+  it("Should open a confirmation dialog when clicked", () => {
+    // check that open is false to begin with
+    expect(setOpen).toBeFalsy;
```

Comment:
Any particular reasonw e're using thruthy & falsy instead of true and false?

Reply:
Not related to my PR, so it would be better to address this issue in @gdzooks's #248 PR.

Reply:
yeah, setOpen is a function, so it can't be a boolean value. truthy/false checks that a function is returning values that qualify as true/false respectively.